md-dao/src/main/java/com/wipro/ats/bdre/md/dao/JobDAO.java

```diff
@@ -135,20 +142,20 @@
             lastRunProcess.setProcessId(processId);
             ExecStatus successExecState = new ExecStatus();
             successExecState.setExecStateId(3);
-            Criteria lastRunFailedCriteria = session.createCriteria(InstanceExec.class).add(Restrictions.eq("process", lastRunProcess)).addOrder(Order.desc("endTs"));
-            if (lastRunFailedCriteria.list().size() > 0) {
+            Criteria lastRunFailedCriteria = session.createCriteria(InstanceExec.class).add(Restrictions.eq(PROCESS, lastRunProcess)).addOrder(Order.desc(ENDTS));
+            if (!lastRunFailedCriteria.list().isEmpty()) {
                 lastRunFailedCriteria.setMaxResults(1);
                 InstanceExec lastRunExecStatus = (InstanceExec) lastRunFailedCriteria.list().get(0);
                 if (lastRunExecStatus.getExecStatus().getExecStateId() == 6) {
-                    Criteria getSuccessfulStepsCriteria = session.createCriteria(InstanceExec.class).add(Restrictions.eq("execStatus", successExecState));
+                    Criteria getSuccessfulStepsCriteria = session.createCriteria(InstanceExec.class).add(Restrictions.eq(EXECSTATUS, successExecState));
                     for (Process stepsProcess : listOfSubProcesses) {
-                        getSuccessfulStepsCriteria.add(Restrictions.eq("process", stepsProcess));
+                        getSuccessfulStepsCriteria.add(Restrictions.eq(PROCESS, stepsProcess));
                     }
-                    getSuccessfulStepsCriteria.addOrder(Order.desc("endTs"));
+                    getSuccessfulStepsCriteria.addOrder(Order.desc(ENDTS));
 
                     for (Object recoverableObject : getSuccessfulStepsCriteria.list()) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)